src/main/java/org/jabref/logic/importer/fetcher/CiteSeer.java

```diff
@@ -46,52 +40,40 @@ public Optional<HelpFile> getHelpPage() {
-        // So we extract the data string from the <span class="Z3988" title="<data>"></span> tags and pass the content to the COinS parser
-        return inputStream -> {
-            String response = new BufferedReader(new InputStreamReader(inputStream)).lines().collect(Collectors.joining(OS.NEWLINE));
-            List<BibEntry> entries = new ArrayList<>();
-            CoinsParser parser = new CoinsParser();
-            Pattern pattern = Pattern.compile("<span class=\"Z3988\" title=\"(.*)\"></span>");
-            Matcher matcher = pattern.matcher(response);
-            while (matcher.find()) {
-                String encodedDataString = matcher.group(1);
-                entries.addAll(parser.parseEntries(encodedDataString));
-            }
-            return entries;
-        };
-    }
+            Optional<JSONArray> jsonResponse = Optional.of(requestResponse)
+                                                    .map(JsonNode::getObject)
+                                                    .filter(Objects::nonNull)
+                                                    .map(response -> response.optJSONArray("response"))
+                                                    .filter(Objects::nonNull);
 
-    @Override
-    public void doPostCleanup(BibEntry entry) {
-        // CiteSeer escapes some characters in a way that is not recognized by the normal html to unicode formatter
-        // We, of course, also want to convert these special characters
-        Formatter extendedHtmlFormatter = new HtmlToUnicodeFormatter() {
-            @Override
-            public String format(String fieldText) {
-                String formatted = super.format(fieldText);
-                formatted = formatted.replaceAll("%3A", ":");
-                formatted = formatted.replaceAll("%3Cem%3", "");
-                formatted = formatted.replaceAll("%3C%2Fem%3E", "");
-                formatted = formatted.replaceAll("%2C\\+", " ");
-                formatted = formatted.replaceAll("\\+", " ");
-                return formatted;
+            if (!jsonResponse.isPresent()) {
+                return List.of();
             }
-        };
-        new FieldFormatterCleanup(InternalField.INTERNAL_ALL_FIELD, extendedHtmlFormatter).cleanup(entry);
 
-        // Many titles in the CiteSeer database have all-capital titles, for convenience we convert them to title case
-        new FieldFormatterCleanup(StandardField.TITLE, new TitleCaseFormatter()).cleanup(entry);
+            CiteSeerParser parser = new CiteSeerParser();
+            List<BibEntry> fetchedEntries = parser.parseCiteSeerResponse(jsonResponse.orElse(new JSONArray()));
+            return fetchedEntries;
+        } catch (ParseException ex) {
+            throw new FetcherException("An internal parser error occurred while parsing CiteSeer entries, ", ex);
+        }
+    }
+
+    private JSONElement getPayloadJSON(QueryNode luceneQuery) {
+        // use CiteSeerQueryTransformer
```

Comment:
I think this comment is extra, as it's obvious from the next line

Reply:
I agree, I think that was for myself before I implemented the QueryTransformer. Will remove.